cmd/agent/app/options/options.go

```diff
@@ -122,6 +129,8 @@ func (o *GrpcProxyAgentOptions) Flags() *pflag.FlagSet {
 	flags.BoolVar(&o.WarnOnChannelLimit, "warn-on-channel-limit", o.WarnOnChannelLimit, "Turns on a warning if the system is going to push to a full channel. The check involves an unsafe read.")
 	flags.BoolVar(&o.SyncForever, "sync-forever", o.SyncForever, "If true, the agent continues syncing, in order to support server count changes.")
 	flags.IntVar(&o.XfrChannelSize, "xfr-channel-size", 150, "Set the size of the channel for transferring data between the agent and the proxy server.")
+	flags.StringVar(&o.ServerLeaseSelector, "server-lease-selector", o.ServerLeaseSelector, "Providing a label selector enables updating the server count by counting the number of valid leases matching the selector.")
```

Comment:
We should probably provide a warning here. Maybe it can be a reference to a section in the OSS docs. Reads should not require a webhook call from the KAS. However writing a lease can require an admission webhook call. This means that attempts to write the lease can fail because there is no tunnel from the KAS to cluster. There may be no tunnel to the cluster because the server count is 0. The server count is 0 because nothing can write a lease.

Reply:
Wondering if we should always assume a lower bound of 1.

Reply:
Current behavior pretty much assumes a lower bound of 1, as a connection is always attempted at first regardless of the server count and the server count is only taken in to consideration if it is != 0:

https://github.com/carreter/apiserver-network-proxy/blob/server-counter-e2e/pkg/agent/clientset.go#L191-L217